File: code-examples/php/crud/text_search.php
<?php

require_once __DIR__ . "/vendor/autoload.php";

// Replace the following string with your MongoDB deployment's connection string
$uri = "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority";
$client = new MongoDB\Client($uri);

$db = $client->sample_mflix;
$movies = $db->movies;

function searchWord($movies) {
    // Create a query that searches for the string "trek"
    $filter = ['\$text' => ['$search' => "trek"]];
    $options = ['projection' => ['_id' => 0, 'title' => 1]];
    $cursor = $movies->find($filter, $options);
    if (count($cursor) === 0) {
        echo "No documents found!";
    }
    foreach ($cursor as $document) {
        var_dump($document);
    }
}

function searchPhrase($movies) {
    // Create a query that searches for the phrase "star trek"
    $filter = ['\$text' => ['$search' => "\"star trek\""]];
    $options = ['projection' => ['_id' => 0, 'title' => 1]];
    $cursor = $movies->find($filter, $options);
    if (count($cursor) === 0) {
        echo "No documents found!";
    }
    foreach ($cursor as $document) {
        var_dump($document);
    }
}

function searchNegation($movies) {
    // Create a query that searches for the phrase "star trek" while omitting "into darkness"
    $filter = ['\$text' => ['$search' => "\"star trek\"  -\"into darkness\""]];
    $options = ['projection' => ['_id' => 0, 'title' => 1]];
    $cursor = $movies->find($filter, $options);
    if (count($cursor) === 0) {
        echo "No documents found!";
    }
    foreach ($cursor as $document) {
        var_dump($document);
    }
}

function relevance($movies) {
    $filter = ['\$text' => ['$search' => "\"star trek\"  -\"into darkness\""]];
    $options = ['projection' => ['_id' => 0, 'title' => 1, 'score' => ['$meta' => "textScore"]],
                'sort' => ['score' => ['$meta' => "textScore"]]];
    $cursor = $movies->find($filter, $options);
    if (count($cursor) === 0) {
        echo "No documents found!";
    }
    foreach ($cursor as $document) {
        var_dump($document);
    }
}

try {
    searchWord($movies);
    searchPhrase($movies);
    searchNegation($movies);
    relevance($movies);
} catch (Exception $e) {
    echo $e->getMessage();
}
?>

File: code-examples/scala/crud/text_search.scala
import org.mongodb.scala._
import scala.concurrent.ExecutionContext.Implicits.global
import scala.util.{Failure, Success}

object MongoDBSearch {

  val mongoClient: MongoClient = MongoClient("mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority")
  val database: MongoDatabase = mongoClient.getDatabase("sample_mflix")
  val movies: MongoCollection[Document] = database.getCollection("movies")

  def searchWord(): Unit = {
    val query: Document = Document("$text" -> Document("$search" -> "trek"))
    val projection: Document = Document("_id" -> 0, "title" -> 1)
    val cursor: FindObservable[Document] = movies.find(query).projection(projection)

    cursor.toFuture().onComplete {
      case Failure(e) => e.printStackTrace()
      case Success(results) => results.foreach(println)
    }
  }

  def searchPhrase(): Unit = {
    val query: Document = Document("$text" -> Document("$search" -> "\"star trek\""))
    val projection: Document = Document("_id" -> 0, "title" -> 1)
    val cursor: FindObservable[Document] = movies.find(query).projection(projection)

    cursor.toFuture().onComplete {
      case Failure(e) => e.printStackTrace()
      case Success(results) => results.foreach(println)
    }
  }

  def searchNegation(): Unit = {
    val query: Document = Document("$text" -> Document("$search" -> "\"star trek\" -\"into darkness\""))
    val projection: Document = Document("_id" -> 0, "title" -> 1)
    val cursor: FindObservable[Document] = movies.find(query).projection(projection)

    cursor.toFuture().onComplete {
      case Failure(e) => e.printStackTrace()
      case Success(results) => results.foreach(println)
    }
  }

  def searchRelevance(): Unit = {
    val query: Document = Document("$text" -> Document("$search" -> "\"star trek\" -\"into darkness\""))
    val sort: Document = Document("score" -> Document("$meta" -> "textScore"))
    val projection: Document = Document("_id" -> 0, "title" -> 1, "score" -> Document("$meta" -> "textScore"))
    val cursor: FindObservable[Document] = movies.find(query).sort(sort).projection(projection)

    cursor.toFuture().onComplete {
      case Failure(e) => e.printStackTrace()
      case Success(results) => results.foreach(println)
    }
  }

  def main(args: Array[String]): Unit = {
    searchWord()
    searchPhrase()
    searchNegation()
    searchRelevance()

    // Close the connection to the MongoDB instance
    mongoClient.close()
  }
}
